sklearn/manifold/_utils.pyx

```diff
@@ -118,4 +116,4 @@ cpdef cnp.ndarray[cnp.float32_t, ndim=2] _binary_search_perplexity(
     if verbose:
         print("[t-SNE] Mean sigma: %f"
               % np.mean(math.sqrt(n_samples / beta_sum)))
-    return P
+    return np.array(P, dtype=np.float32)
```

Comment:
I think preserving dtype might better be treated in another PR.

For now, I propose just returning the numpy array using the `base` attribute of the memoryview.
```suggestion
    return P.base
```

Reply:
It was discussed somewhere that np.asarray should be preferred because `.base` can point to a larger array than expected (we can have a view on only a subset of the array)